pkg/sql/rename_database.go

```diff
@@ -161,6 +198,79 @@ func (n *renameDatabaseNode) startExec(params runParams) error {
 	return p.renameDatabase(ctx, dbDesc, n.newName)
 }
 
+// isAllowedDependentDescInRename determines rename database is allowed when
+// a given {tbDesc, dependentDesc} with the relationship dependedOn on a db named dbName.
+// Returns a bool representing whether it's allowed, a string indicating the column name
+// it could not found, and an error if there is any.
+// This is a workaround for #45411 until #34416 is resolved.
+func isAllowedDependentDescInRenameDatabase(
+	dependedOn sqlbase.TableDescriptor_Reference,
+	tbDesc *sqlbase.TableDescriptor,
+	dependentDesc *sqlbase.TableDescriptor,
+	dbName string,
+) (bool, string, error) {
+	// If it is a sequence, and it does not contain the database name, then we have
+	// no reason to block it's deletion.
+	if !tbDesc.IsSequence() {
+		return false, "", nil
+	}
+
+	colIDs := util.MakeFastIntSet()
+	for _, colID := range dependedOn.ColumnIDs {
+		colIDs.Add(int(colID))
+	}
+
+	for _, column := range dependentDesc.Columns {
+		if !colIDs.Contains(int(column.ID)) {
+			continue
+		}
+		colIDs.Remove(int(column.ID))
+
+		if column.DefaultExpr == nil {
+			return false, "", errors.AssertionFailedf(
+				"rename_database: expected column id %d in table id %d to have a default expr",
+				dependedOn.ID,
+				dependentDesc.ID,
+			)
+		}
+		// Try parse the default expression and find the table name direct reference.
+		parsedExpr, err := parser.ParseExpr(*column.DefaultExpr)
+		if err != nil {
+			return false, "", err
+		}
+		typedExpr, err := tree.TypeCheck(parsedExpr, nil, &column.Type)
+		if err != nil {
+			return false, "", err
+		}
+		seqNames, err := getUsedSequenceNames(typedExpr)
+		if err != nil {
+			return false, "", err
+		}
+		for _, seqName := range seqNames {
+			parsedSeqName, err := parser.ParseTableName(seqName)
+			if err != nil {
+				return false, "", err
+			}
+			// There must be at least two parts for this to work.
+			if parsedSeqName.NumParts >= 2 {
+				// We only don't allow this if the database name is in there.
+				// This is always the last argument.
+				if tree.Name(parsedSeqName.Parts[parsedSeqName.NumParts-1]).Normalize() == tree.Name(dbName).Normalize() {
+					return false, column.Name, nil
```

Comment:
Is this really the column name we couldn't find? It seems more like the column that has the db name reference.